### ISSUE TITLE
Logging sinks resources do not depend on IAM rules

### ISSUE THREAD

**Issue by quulah** [terraform-google-modules/cloud-foundation-fabric#371]
### TL;DR

When setting both IAM roles and logging sinks for a folder, the sinks are created before authoritative role attachments.

If the role happens to be `roles/logging.admin` for example and the user running Terraform is in the group, the sink creation fails.

### Expected behavior

Creating IAM rules and other resources should work in one go.

### Observed behavior

Creation of resources, dependent on the IAM rules fails.

### Terraform Configuration

```hcl
module "testing_folder" {
  source = "github.com/terraform-google-modules/cloud-foundation-fabric//modules/folder?ref=v8.0.0"

  parent = var.root_node
  name   = "testing"

  group_iam = {
    "masters-of-log@example.com" = [
       "roles/logging.admin"
    ]
  }

  logging_sinks = {
    testing = {
      type = "logging"

      destination = "a-bucket-in-a-project"

      filter     = <<-EOT
          NOT LOG_ID("cloudaudit.googleapis.com/activity") AND
          NOT LOG_ID("externalaudit.googleapis.com/activity") AND
          NOT LOG_ID("cloudaudit.googleapis.com/system_event") AND
          NOT LOG_ID("externalaudit.googleapis.com/system_event") AND
          NOT LOG_ID("cloudaudit.googleapis.com/access_transparency") AND
          NOT LOG_ID("externalaudit.googleapis.com/access_transparency")
      EOT
      exclusions = {}

      iam = true
      include_children = true
    }
  }
```

### Terraform Version

```sh
Terraform v1.0.10
on darwin_arm64
+ provider registry.terraform.io/aiven/aiven v2.3.2
+ provider registry.terraform.io/hashicorp/google v4.0.0
+ provider registry.terraform.io/hashicorp/google-beta v4.0.0

Your version of Terraform is out of date! The latest version
is 1.0.11. You can update by downloading from https://www.terraform.io/downloads.html
```

### Additional information

I'm guessing this type of issue might be hitting other modules (organization, project) and other resources as well. I just happened to stumble upon it when creating some logging sinks.

**Comment by ludoo** [comment]
Hey Miika, good catch. We can probably address this by depending the resource on the one applying IAM roles. I'll try and send a patch shortly.

**Comment by ludoo** [comment]
@quulah can you check if the above merge fixes your issue? We'll release a patch version ASAP if it does.

**Comment by quulah** [comment]
Thanks @ludoo!

I tested with a new folder, IAM binding and logging sink. Seems to be working. Or at least the IAM things get added prior to the sink creation being attempted.

My try actually resulted in the project for the sink taking a while, so it actually failed once more. But I trust your changes. 🙈

**Comment by quulah** [comment]
I encountered a related thing, since accidentally removed the IAM binding first 🙄.

If the sink already exists, and the necessary IAM binding does not, it'll fail in `terraform plan` phase when refreshing the state of the sink. Even though you'd be adding the binding back with `iam` or `group_iam` at the same time.

But, I don't think we can actually do anything about that.
